currency-rates/internal/repo/subsctiption.go

```diff
@@ -12,27 +12,28 @@ type DBEmail struct {
 }
 
 func (repo *Repository) SetEmail(ctx context.Context, email string) error {
-	e := &DBEmail{
-		Email: email,
-		IsActive: true,
-	}
-
-	return repo.DB.Table("emails").Create(e).Error
+    e := &DBEmail{
+        Email: email,
+        IsActive: true,
+    }
+
+    return repo.metrics.MonitorDBQuery("SetEmail", func() error {
```

Comment:
I don't like this way of injecting metrics specific into repo logic. You could use such ways like 'Decorator' 'Middelware' etc. for this, not straight dependency and injection for your db.